libs/utils/utils.js

```diff
@@ -155,6 +155,143 @@ function decoratePictures(el) {
+    let areaIndex = 0;
+    if (icons[0].classList.contains('icon-persona')) {
+      icons[0].closest('p').classList.add('persona-area');
+      areaIndex = 1;
+    }
+    icons[areaIndex]?.closest('p').classList.add('icon-area');
+  }
+}
+
+export function decorateBlockDaa(el) {
+  const lh = [];
+  const exclude = ['--', 'block'];
+  el.classList.forEach((c) => {
+    if (!c.includes(exclude[0]) && c !== exclude[1]) lh.push(c);
+  });
+  el.setAttribute('daa-im', 'true');
+  el.setAttribute('daa-lh', lh.join('|'));
+}
+
+export function decorateTextDaa(el, heading) {
+  el.setAttribute('daa-lh', heading.textContent);
+  const links = el.querySelectorAll('a, button');
+  if (links) {
+    links.forEach((link, i) => {
+      const linkType = () => {
+        if (link.classList.contains('con-button')) {
+          return 'cta';
+        }
+        if (link.classList.contains('icon')) {
+          return 'icon cta';
+        }
+        return 'link';
+      };
+      const str = `${linkType(link)}|${link.innerText} ${i + 1}`;
+      link.setAttribute('daa-ll', str);
+    });
+  }
+}
+
+export function decorateText(el, size) {
+  const headings = el.querySelectorAll('h1, h2, h3, h4, h5, h6');
+  const heading = headings[headings.length - 1];
+  if (!size || size === 'small') {
+    heading.classList.add('heading-XS');
+    heading.nextElementSibling.classList.add('body-S');
+    if (heading.previousElementSibling) {
+      heading.previousElementSibling.classList.add('detail-M');
+    }
+  }
+  if (size === 'medium') {
+    heading.classList.add('heading-M');
+    heading.nextElementSibling.classList.add('body-S');
+    if (heading.previousElementSibling) {
+      heading.previousElementSibling.classList.add('detail-M');
+    }
+  }
+  if (size === 'large') {
+    heading.classList.add('heading-XL');
+    heading.nextElementSibling.classList.add('body-M');
+    if (heading.previousElementSibling) {
+      heading.previousElementSibling.classList.add('detail-L');
+    }
+  }
+  decorateIcons(el);
+  decorateButtons(el);
+  decorateTextDaa(el, heading);
+}
+
+export function isHexColorDark(color) {
+  if (color[0] !== '#') return false;
+  const hex = color.replace('#', '');
+  const cR = parseInt(hex.substr(0, 2), 16);
+  const cG = parseInt(hex.substr(2, 2), 16);
+  const cB = parseInt(hex.substr(4, 2), 16);
+  const brightness = ((cR * 299) + (cG * 587) + (cB * 114)) / 1000;
+  return brightness < 155;
+}
+
+export function decorateBlockBg(block, node) {
+  node.classList.add('background');
+  if (!node.querySelector(':scope img')) {
+    block.style.background = node.textContent;
+    if (isHexColorDark(node.textContent)) block.classList.add('dark');
+    node.remove();
+  }
+}
+
+export function getBlockSize(el) {
+  const sizes = ['small', 'medium', 'large'];
+  let defaultSize = sizes[1];
+  sizes.forEach((size) => {
+    if (el.classList.contains(size)) {
+      defaultSize = size;
+    }
+  });
+  return defaultSize;
```

Comment:
Seems weird to redefine default size.

I **_think_** this should work...

```suggestion
export function getBlockSize(el) {
  const sizes = ['small', 'medium', 'large'];
  return sizes.reduce((rdx, size) => {
    if (el.classList.contains(size)) {
      return size;
    }
    return rdx;
  }, sizes[1]);
}
```

@chrischrischris may have something better.